torch/nn/modules/adaptive.py

```diff
@@ -273,7 +273,9 @@ def _get_full_log_prob(self, input, head_output):
 
         out[:, : self.shortlist_size] = head_logprob[:, : self.shortlist_size]
 
-        for i, (start_idx, stop_idx) in enumerate(zip(self.cutoffs, self.cutoffs[1:])):
+        for i, (start_idx, stop_idx) in enumerate(
+            zip(self.cutoffs, self.cutoffs[1:], strict=True)
```

Comment:
This should actually itertools pairwise? And it will fail when string is True

Reply:
Yes, I should fix it.